core/templates/components/forms/schema-based-editors/schema-based-choices-editor.component.spec.ts

```diff
@@ -40,34 +38,58 @@ describe('Schema Based Choices Editor Component', () => {
     component = fixture.componentInstance;
 
     component.ngOnInit();
-    component.registerOnTouched();
-    component.registerOnChange(null);
-    component.onChange = (val: boolean) => {
-      return;
-    };
   });
 
+  it('should set component properties on initialization', fakeAsync(() => {
+    let mockFunction = function(value: number) {
+      return value;
+    };
+    component.registerOnChange(mockFunction);
+    component.registerOnTouched();
+
+    expect(component).toBeDefined();
+    expect(component.validate(null)).toEqual({});
+    expect(component.onChange).toEqual(mockFunction);
+    expect(component.onChange(true)).toEqual(true);
+  }));
+
   it('should get empty object on validating', () => {
     expect(component.validate(null)).toEqual({});
   });
 
-  it('should overwrite local value', () => {
-    expect(component.localValue).toBe(undefined);
+  it('should write value', () => {
+    component.localValue = null;
+    component.writeValue(null);
 
-    component.writeValue(true);
+    expect(component.localValue).toEqual(null);
 
+    component.writeValue(true);
     expect(component.localValue).toBeTrue();
   });
 
-  it('should update local value', () => {
-    component.localValue = false;
+  it('should update value when local value change', () => {
+    component.localValue = true;
```

Comment:
I don't think local value can be boolean in this case. At the very least it should be an Array. I think an Array of strings should be a good guess. You can look at the existing tests in develop as a base. For this and all other tests.

Reply:
I'm unable to understand, why array of string, we uses type checkbox's or option's in html which means updated value change inbetween true and false. And also in component.ts argument of update function is type of boolean.